tests/executionspec/spec.rs

```diff
@@ -0,0 +1,403 @@
+extern crate core;
+
+use std::collections::BTreeMap;
+use std::path::Path;
+use std::sync::Arc;
+use std::{fs, panic};
+
+use anyhow::Context;
+use futures_util::future::join_all;
+use hyper::{Body, Request};
+use serde::{Deserialize, Serialize};
+use tailcall::async_graphql_hyper::{GraphQLBatchRequest, GraphQLRequest};
+use tailcall::blueprint::Blueprint;
+use tailcall::config::reader::ConfigReader;
+use tailcall::config::{Config, ConfigModule, Source};
+use tailcall::http::{handle_request, AppContext};
+use tailcall::merge_right::MergeRight;
+use tailcall::print_schema::print_schema;
+use tailcall::valid::{Cause, ValidationError, Validator as _};
+use telemetry::in_memory::InMemoryTelemetry;
+use telemetry::init::init_opentelemetry;
+
+use super::parse::ExecutionSpec;
+use super::runtime::{APIRequest, APIResponse, MockFileSystem, MockHttpClient};
+use crate::telemetry;
+
+#[derive(Serialize, Deserialize, Clone, Debug)]
+#[serde(rename_all = "camelCase")]
+enum Annotation {
+    Skip,
+    Only,
+}
+
+#[derive(Debug, Default, Deserialize, Serialize, PartialEq)]
+struct SDLError {
+    message: String,
+    trace: Vec<String>,
+    description: Option<String>,
+}
+
+impl<'a> From<Cause<&'a str>> for SDLError {
+    fn from(value: Cause<&'a str>) -> Self {
+        SDLError {
+            message: value.message.to_string(),
+            trace: value.trace.iter().map(|e| e.to_string()).collect(),
+            description: None,
+        }
+    }
+}
+
+impl From<Cause<String>> for SDLError {
+    fn from(value: Cause<String>) -> Self {
+        SDLError {
+            message: value.message.to_string(),
+            trace: value.trace.iter().map(|e| e.to_string()).collect(),
+            description: value.description,
+        }
+    }
+}
+
+async fn test_spec(spec: ExecutionSpec, opentelemetry: &InMemoryTelemetry) {
```

Comment:
could we split this big function into smaller pieces to make it more readable?

Reply:
will do it.